File: models/farm.go
package models

import (
	"encoding/json"
	"time"

	"github.com/gobuffalo/pop/v6"
	"github.com/gobuffalo/validate/v3"
	"github.com/gobuffalo/validate/v3/validators"
	"github.com/gofrs/uuid"
)

// Farm is used by pop to map your Farms database table to your go code.
type Farm struct {
	ID            uuid.UUID `json:"id" db:"id"`
	Name          string    `json:"name" db:"name"`
	Description   string    `json:"description" db:"description"`
	AccountID     uuid.UUID `json:"account_id" db:"account_id"`
	Account       *Account  `json:"Account,omitempty" belongs_to:"account"`
	CreatedAt     time.Time `json:"created_at" db:"created_at"`
	UpdatedAt     time.Time `json:"updated_at" db:"updated_at"`
}


// String is not required by pop and may be deleted
func (f Farm) String() string {
	jf, _ := json.Marshal(f)
	return string(jf)
}

// Farms is not required by pop and may be deleted
type Farms []Farm

// String is not required by pop and may be deleted
func (f Farms) String() string {
	jf, _ := json.Marshal(f)
	return string(jf)
}

// Validate gets run every time you call a "pop.Validate*" (pop.ValidateAndSave, pop.ValidateAndCreate, pop.ValidateAndUpdate) method.
// This method is not required and may be deleted.
func (f *Farm) Validate(tx *pop.Connection) (*validate.Errors, error) {
	return validate.Validate(
		&validators.StringIsPresent{Field: f.Name, Name: "Name"},
	), nil
}

// ValidateCreate gets run every time you call "pop.ValidateAndCreate" method.
// This method is not required and may be deleted.
func (f *Farm) ValidateCreate(tx *pop.Connection) (*validate.Errors, error) {
	return validate.NewErrors(), nil
}

// ValidateUpdate gets run every time you call "pop.ValidateAndUpdate" method.
// This method is not required and may be deleted.
func (f *Farm) ValidateUpdate(tx *pop.Connection) (*validate.Errors, error) {
	return validate.NewErrors(), nil
}

File: frontend/lib/model/farm.dart
class Farm {
	final String id;
	final String name;
  	final String description;
	final String account_id;
	final DateTime createdAt;
	final DateTime updatedAt;

  	Farm({required this.id, required this.name, required this.description, required this.account_id, required this.createdAt, required this.updatedAt});

  	factory Farm.fromJson(Map<String, dynamic> json) {
    		return Farm(
      			id: json['id'],
      			name: json['name'],
      			description: json['description'],
			account_id: json['account_id'],
			createdAt: DateTime.parse(json['created_at']),
			updatedAt: DateTime.parse(json['updated_at']),
    		);
  	}

  	Map<String, dynamic> toJson() {
    		return {
      			'id': id,
      			'name': name,
      			'description': description,
			'account_id': account_id,
			'created_at': createdAt.toIso8601String(),
			'updated_at': updatedAt.toIso8601String(),
    		};
  	}
}
